### ISSUE TITLE
Compilation fails with "Computation name is not unique: add_computation" when using multiple np.sum operations in a kernel

### ISSUE THREAD

**Issue by AoyuQC** [aws-neuron/nkipy#1]
When a kernel contains multiple np.sum operations (e.g., softmax followed by another sum for normalization), the HLO lowering produces duplicate computation names, causing the Neuron compiler to fail with:

`Computation name is not unique: add_computation`

This appears to be a bug in how NKIPy's HLO lowering generates computation names for reduction operations.

## Environment

* NKIPy version: latest
* Neuron SDK: 2.22.12471.0+b4a00d10
* Instance type: trn2.48xlarge

## Reproduction

The following minimal example demonstrates the issue. A router kernel that performs:

1. softmax (which internally uses np.sum)
2. topk
3. Another np.sum to normalize the top-k weights

```
import numpy as np
from ml_dtypes import bfloat16
from nkipy.core.trace import NKIPyKernel
from nkipy.runtime.device_kernel import DeviceKernel
from nkipy.core import tensor_apis

def softmax(x, is_neuronpy):
    """Softmax with sum - first np.sum usage."""
    x = x.astype(np.float32)
    max_x = np.max(x, axis=-1, keepdims=True)
    exp_x = np.exp(x - max_x)
    sum_exp = np.sum(exp_x, axis=-1, keepdims=True)  # <-- First sum
    return (exp_x / sum_exp).astype(x.dtype)

def router_kernel(hidden_states, router_weight, top_k, is_neuronpy):
    """This kernel FAILS to compile."""
    n_tokens, n_experts = hidden_states.shape[0], router_weight.shape[1]
    
    router_logits = (hidden_states @ router_weight).astype(hidden_states.dtype)
    routing_weights = softmax(router_logits, is_neuronpy)
    
    top_k_weights, top_k_indices = tensor_apis.topk(routing_weights, k=top_k, axis=1)
    top_k_indices = top_k_indices.astype(np.int32)
    
    # THIS CAUSES THE BUG: second np.sum after softmax's internal sum
    top_k_weights = top_k_weights.astype(np.float32)
    weight_sum = np.sum(top_k_weights, axis=-1, keepdims=True)  # <-- Second sum
    top_k_weights = (top_k_weights / weight_sum).astype(hidden_states.dtype)
    
    # ... scatter logic ...
    return top_k_indices, top_k_weights

# Compile - this fails
trace = NKIPyKernel.trace(backend="hlo")
DeviceKernel.compile_and_load(
    trace(router_kernel),
    name="router",
    is_neuronpy=True,
    hidden_states=np.random.randn(32, 256).astype(bfloat16),
    router_weight=np.random.randn(256, 128).astype(bfloat16),
    top_k=8,
)
```

Expected Behavior
Multiple `np.sum` operations in the same kernel should compile successfully. The HLO lowering should generate unique computation names (e.g., `add_computation_0`, `add_computation_1`, etc.).

### Actual Behavior

Compilation fails with:

`Computation name is not unique: add_computation`

### Workarounds

Two workarounds exist, but they shouldn't be necessary:

**Workaround 1**: Use divide-then-multiply pattern to avoid direct np.sum:

```
# Instead of:
weight_sum = np.sum(top_k_weights, axis=-1, keepdims=True)

# Use:
weight_sum = np.divide(top_k_weights, top_k).astype(top_k_weights.dtype)
weight_sum = np.sum(weight_sum, axis=-1, keepdims=True) * top_k
```

**Workaround 2**: Restructure the algorithm to avoid multiple sums (e.g., apply topk before softmax).

## Root Cause Hypothesis

The HLO lowering for np.sum (and likely other reductions) appears to use a hardcoded computation name like add_computation without appending a unique suffix. When multiple sums exist in the same kernel, this creates a name collision.

See [repro_router_compiler_bug.py](https://github.com/user-attachments/files/24699737/repro_router_compiler_bug.py)  in the issue attachments
